[11.0] [FIX] Overeager procurement rule expansion

StefanRijnhart opened the pull request:
When expanding what it considers 'global' procurement rules and location
paths without a route, the stock migration incorrectly also expands items
that already did have a route.

I'm also proposing a different treatment of inactive items, instead of
expanding them in a huge amount of just more inactive items.

Also, by not removing the original records from the database we preserve the
referential links from stock moves and procurement orders to the procurement
rules.

--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

StefanRijnhart commented:
BTW in our setup application management have deleted many routes they deemed incorrect or outdated. That left a lot of dangling, unused procurement rules. I'll be asking them to set those rules to inactive before the migration, and then this code will prevent the expansion.

hbrunn commented on addons/stock/migrations/11.0.1.1/pre-migration.py:
```diff
@@ -2,21 +2,104 @@
+    """
+    dummy_route_id = False
+    env.cr.execute(
+        "SELECT id FROM stock_location_route WHERE active ORDER BY id")
```
shouldn't you first order by [`sequence`](https://github.com/OCA/OpenUpgrade/blob/10.0/addons/stock/models/stock_location.py#L118)?

StefanRijnhart replied:
Ah, the ordering is a remainder from an earlier attempt in which I needed to know in the post script which route I assigned in the pre script. It is arbitrary but in the current version I might just as well leave it out (and I might just as well leave it in, which I think I will).

StefanRijnhart replied:
But I will add a limit

StefanRijnhart replied:
Added a limit, removed the ordering after all and added a clarifying comment.

StefanRijnhart committed:
[FIX] Overeager procurement rule expansion

When expanding what it considers 'global' procurement rules and location
paths without a route, the stock migration incorrectly also expands items
that already did have a route.

I'm also proposing a different treatment of inactive items, instead of
expanding them in a huge amount of just more inactive items.

Also, by not removing the original records from the database we preserve the
referential links from stock moves and procurement orders to the procurement
rules.

MiquelRForgeFlow commented:
@StefanRijnhart @hbrunn could you check https://github.com/OCA/OpenUpgrade/pull/2502? I think my approach it's more readable and cleaner.

StefanRijnhart commented:
Absolutely!
